python/cudf/cudf/core/index.py

```diff
@@ -1808,6 +1971,40 @@ def categories(self):
         """
         return self._values.cat().categories
 
+    @annotate("CATEGORICAL_INDEX_EQUALS", color="green", domain="cudf_python")
+    def equals(self, other):
+        """
+        Determine if two Index objects contain the same elements.
+
+        Returns
+        -------
+        out: bool
+            True if “other” is an Index and it has the same elements
+            as calling index; False otherwise.
+        """
+        if self is other:
+            return True
+        elif not isinstance(other, Index):
+            return False
+        elif len(self) != len(other):
+            return False
+        elif not isinstance(other, Index):
+            return False
+        else:
+            casted_other = other
+            if not is_categorical_dtype(other.dtype):
+                casted_other = other.astype("category")
+                casted_other = casted_other.astype(self.dtype)
+
+            if self.dtype != casted_other.dtype:
+                return False
+
+            result = self._values == casted_other._values
+            if isinstance(result, bool):
+                return result
```

Comment:
When is the the case versus it being an elementwise equals and column return?

Reply:
This was added initially when `CategoricalIndex` case was being handled in a common api, this is the case in `CategoricalIndex` only as there we just need to compare the column.